### ISSUE TITLE
[Dynamism] Running forward pass of FCOS model

### ISSUE THREAD

**Issue by ymwangg** [pytorch/xla#4377]
## 🚀 Feature
We'll use this thread to track issues related to running the forward pass of [FCOS](https://pytorch.org/vision/main/models/fcos.html) model.

## Issues
Tested using this [script](https://gist.github.com/ymwangg/b26146ab640c74a2823df62a95fff56f) with flag `XLA_EXPERIMENTAL="nonzero"`.
1. `torch.where(x > 0.5)` (lowered to `aten::nonzero`) returns wrong output dtype int32 instead of int64.
2. `aten::index` does not support int32 as input element type.
3. `min(SymInt, int)` raised "NYI" error.
4. `torch.topk` does not support SymInt type as input parameter `k`. 
5. unlowered ops: `aten::_unique2`, `torchvision::nms`.

**Comment by JackCaoG** [comment]
`aten::nonzero` return `int32` is expected, on XLA it seems like we expect the size(used for `SetDimensionSize` and return by `GetDimensionSIze`) to be `int32` instead of `int64` for XLA:GPU and XLA:TPU.

**Comment by JackCaoG** [comment]
I explicitly to make size to be int32 in https://github.com/pytorch/xla/pull/4243.. I guess now since `aten::index` does not work with s32 we need a different solution.

**Comment by ymwangg** [comment]
Yes if `aten::index` accepts s32 then issue 1 shoudn't be significant. The native pytorch `aten::index` supports s32 type but torch_xla blocked it [here](https://github.com/pytorch/xla/commit/74c98bb92d6245cb6292d7ef5b2b4586d5df527c). I'll see if I can enable it.
